third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -40,6 +40,12 @@ use std::{
     rc::Rc,
 };
 
+/// Values are recursive, and so it is important that we do not use recursive algorithms over
+/// deeply nested values as it can cause stack overflow. Since it is not always possible to avoid
+/// recursion, we opt for a reasonable limit on VM value depth. It is defined in Move VM config,
+/// but since it is difficult to propagate config context everywhere, we use this constant.
+pub const DEFAULT_MAX_VM_VALUE_NESTED_DEPTH: u64 = 128;
```

Comment:
Don't quite understand this. Are you saying we have the VMConfig, but its ignored here in this module since we do not have the context? Wouldn't it be problematic if those things do not coincide?

Reply:
VM config is in `runtime` crate, and this is types crate, so either I need to change all the callers to pass config (which quickly gets ugly), or use a constant which we can still upgrade with feature gating.

VM config uses the same constant, to make sure when this changes, the config also changes. Added a note on that to the comment.